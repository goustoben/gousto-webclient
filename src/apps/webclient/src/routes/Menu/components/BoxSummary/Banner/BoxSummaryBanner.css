@value basketBarHeight 4rem;
@value BreakpointToMedium from 'design-language/breakpoints.css'; /* imported for isToMedium in BoxSummaryBanner.js */
@value BreakpointFromMedium, BreakpointFromLarge from 'design-language/breakpoints.css';
@value FE-900 #BC1526;
@value FE-800 #CB2432;

.boxSummaryBanner {
  z-index: 500;
  position: fixed;

  left: 0;
  bottom: 0;
  width: 100%;
  height: basketBarHeight;
  box-shadow: 0px -4px 8px rgba(0, 0, 0, 0.16);
  background-color: rgba(51, 61, 71, 0.95);
  backdrop-filter: blur(4px);

  @media BreakpointFromMedium {
    display: flex;
    justify-content: space-around;
    align-items: center;
    z-index: 501;
    padding-left: 0.75rem;
  }

  @media BreakpointFromLarge {
    padding-left: 2rem;
  }
}

.buttonsContainer {
  display: flex;
  justify-content: space-between;
  margin: 0;
  height: 100%;
  align-items: center;

  padding: 0 0.75rem;

  @media BreakpointFromMedium {
    flex: 1;
    justify-content: flex-end;
    padding: 0;
  }
}

.errorTooltip {
  position: fixed;
}

.errorMessage {
  composes: fontStyleBody from 'design-language/typography.css';
  color: FE-900;
  box-shadow: 0px 2px 5px FE-800;
}
