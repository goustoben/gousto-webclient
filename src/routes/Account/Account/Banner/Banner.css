@value White, Blackberry from 'styles/colors.css';
@value sizeXXL from 'styles/typography.css';

.banner {
	background: White;
	color: Blackberry;
	composes: paddingMDTopBottom paddingSMLeftRight from 'sass-loader!styles/spacing.scss';
}

.bannerInner {
	composes: accountContainer from 'styles/layout.css';
}

.onDesktop {
    composes: hidden-sm-down from 'sass-loader!styles/bootstrap.scss';
}

.onMobile {
    composes: hidden-md-up from 'sass-loader!styles/bootstrap.scss';
}

.bannerTitle {
	composes: headerXXL from 'styles/typography.css';
	color: Blackberry;
}
