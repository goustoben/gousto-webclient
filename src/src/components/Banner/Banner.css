@value Bluecheese, White from 'styles/colors.css';

.container {
  composes: paddingSMTopBottom from 'sass-loader!styles/spacing.scss';
  text-align: center;
  background: Bluecheese;
}

.hide {
	composes: none from 'styles/display.css';
}

.text {
  composes: inlineBlock from 'styles/display.css';
  composes: marginXXS from 'sass-loader!styles/spacing.scss';
  composes: textXL bodyFont textBold from 'styles/typography.css';
  color: White;
}

.link {
  cursor: pointer;
}

.linkUnderlined {
  text-decoration: underline;
}
