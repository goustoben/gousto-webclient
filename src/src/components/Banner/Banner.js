import PropTypes from 'prop-types'
import React from 'react'
import classnames from 'classnames'
import { Button } from 'goustouicomponents'
import { onEnter } from 'utils/accessibility'
import css from './Banner.css'

const Banner = ({ hide, onClick, text, linkText, fixed, fixedText }) => (
  <div
    role="button"
    onClick={() => { onClick() }}
    onKeyDown={(e) => { onEnter(e, onClick) }}
    tabIndex={0}
    className={classnames(
      css.container,
      css.link,
      { [css.hide]: hide },
      { [css.fixed]: fixed }
    )}
  >
    <p className={css.text}>{fixed ? fixedText : text}&nbsp;</p>
    <Button color="tertiary">{linkText}</Button>
  </div>
)

Banner.propTypes = {
  hide: PropTypes.bool,
  onClick: PropTypes.func,
  text: PropTypes.string,
  linkText: PropTypes.string,
  fixed: PropTypes.bool,
  fixedText: PropTypes.string,
}

Banner.defaultProps = {
  fixed: false,
  fixedText: '',
}

export default Banner
