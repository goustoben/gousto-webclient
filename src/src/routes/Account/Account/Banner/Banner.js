import PropTypes from 'prop-types'
import React from 'react'
import classnames from 'classnames'
import css from './Banner.css'

const Banner = ({ title, version }) => (
	<div className={classnames(css.banner, version ? css[`on${version}`] : '')}>
		<div className={css.bannerInner}>
			<h1 className={css.bannerTitle}>{title}</h1>
		</div>
	</div>
)

Banner.propTypes = {
  title: PropTypes.string,
  version: PropTypes.string,
}

Banner.defaultProps = {
  title: '',
  version: 'Desktop',
}

export default Banner
