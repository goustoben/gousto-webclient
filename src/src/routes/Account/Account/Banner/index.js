export { default as Component } from './Banner'
