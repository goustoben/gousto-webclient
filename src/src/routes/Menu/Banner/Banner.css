@value ScreenXXSMin, ScreenSMMin, ScreenMDMin, ScreenMLMin from 'styles/screenSizes.css';

@value JWBlue #3B3FB2;
@value JWSMImageSize 135px;
@value LGImageSize 200px;
@value MDImageSize 175px;
@value SMImageOffset 0.6;
@value LGImageOffset 0.9;
@value GelSMLeft 50%;
@value GelMDLeft 90%;
@value GelLGLeft 85%;

.container {
	composes: inlineBlock from 'styles/display.css';
	width: 100%;
	height: 200px;

	background-position: center;
	background-size: cover;
}

.wagamama2 {
	composes: container;
	background-image: url('media/images/menu/wagamama_banner_mobile_button.jpg');

	@media ScreenXXSMin {
		background-image: url('media/images/menu/wagamama_banner_tablet_button.jpg');
	}

	@media ScreenMLMin {
		background-image: url('media/images/menu/wagamama_banner_desktop_button.jpg');
	}
}

.christmas1 {
	composes: container;
	background-image: url('media/images/menu/christmas_banner_mobile_no_click.jpg');

	@media ScreenXXSMin {
		background-image: url('media/images/menu/christmas_banner_tablet_no_click.jpg');
	}

	@media ScreenMLMin {
		background-image: url('media/images/menu/christmas_banner_desktop_no_click.jpg');
	}
}

.christmas2 {
	composes: container;
	background-image: url('media/images/menu/christmas_banner_mobile_click.jpg');

	@media ScreenXXSMin {
		background-image: url('media/images/menu/christmas_banner_tablet_click.jpg');
	}

	@media ScreenMLMin {
		background-image: url('media/images/menu/christmas_banner_desktop_click.jpg');
	}
}

.contentBase {
	composes: flex from 'styles/flex.css';
}

.content {
	composes: contentBase;
	position: relative;
	max-width: 1600px;
	margin: 0 auto;
	height: 100%;
}

.gelPortrait {
	composes: gel from 'Gel/Gel.css';
	left: calc(GelSMLeft - (SMImageOffset * JWSMImageSize));
	top: 50%;
	height: JWSMImageSize;
	width: JWSMImageSize;

	@media ScreenSMMin {
		left: calc(GelMDLeft - (SMImageOffset * MDImageSize));
		height: MDImageSize;
		width: MDImageSize;
	}

	@media ScreenMDMin {
		left: calc(GelLGLeft - (LGImageOffset * LGImageSize));
		height: LGImageSize;
		width: LGImageSize;
	}
}

.gelPortrait__image {
  border-radius: 100%;
}

.link {
	width: 100%;
}
