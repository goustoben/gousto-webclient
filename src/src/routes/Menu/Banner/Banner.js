import PropTypes from 'prop-types';
import React from 'react';

import config from 'config/menu'
import BoostAndBalanceBanner from '../BoostAndBalanceBanner'
import FineDineInBanner from '../FineDineInBanner'
import TenToTableBanner from '../TenToTableBanner'

const tenToTableValid = () => (
	new Date() > new Date(config.tenToTableBanner.startDate)
)

const Banner = ({ isAuthenticated }) => {
	if (tenToTableValid()) {
		return <TenToTableBanner />
	}

	return (isAuthenticated) ? <FineDineInBanner /> : <BoostAndBalanceBanner />
}

Banner.propTypes = {
	isAuthenticated: PropTypes.bool,
}

Banner.defaultProps = {
	isAuthenticated: false,
}

export default Banner
