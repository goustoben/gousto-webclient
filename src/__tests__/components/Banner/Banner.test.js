import React from 'react'
import { shallow } from 'enzyme'
import Banner from 'Banner'
import renderer from 'react-test-renderer'

describe('Banner', () => {
  describe('Initial Render', () => {
    test('should return a div', () => {
      const wrapper = shallow(<Banner />)
      expect(wrapper.type()).toEqual('div')
    })

    test('should contain a paragraph', () => {
      const wrapper = shallow(<Banner />)
      expect(wrapper.find('p').length).toEqual(1)
    })

    test('should render a button', () => {
      const wrapper = shallow(<Banner />)
      expect(wrapper.find('Button').length).toEqual(1)
    })

    test('should render text if text prop is set', () => {
      const text = 'Fusce a quam'
      const wrapper = shallow(<Banner text={text} />)
      expect(wrapper.find('p').text()).toContain(text)
    })

    test('should render with given text', () => {
      const text = "A Banner's text"
      const tree = renderer.create(<Banner text={text} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('Alternative Render', () => {
    test('should hide banner if hidden is true', () => {
      const wrapper = shallow(<Banner hide />)
      expect(wrapper.find('div').hasClass('hide')).toBe(true)
    })
  })

  describe('Functionality', () => {
    test('should trigger passed onClick function when clicked', () => {
      const clickFunMock = jest.fn()
      const wrapper = shallow(<Banner onClick={clickFunMock} />)
      wrapper.simulate('click')
      expect(clickFunMock).toHaveBeenCalledTimes(1)
    })
  })
})
