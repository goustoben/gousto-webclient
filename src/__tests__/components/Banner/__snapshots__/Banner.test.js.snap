// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Initial Render should render with given text 1`] = `
<div
  className="container link"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <p
    className="text"
  >
    A Banner's text
    Â 
  </p>
  <div
    className="container tertiary fill widthAuto"
  >
    <div
      className="spinnerContainer  base tertiary fill noHover autoWidthPadding"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <span
        className="spinner"
      >
        <div
          className="fadingCircle"
        >
          <div
            className="circle1 white"
          />
          <div
            className="circle2 white"
          />
          <div
            className="circle3 white"
          />
          <div
            className="circle4 white"
          />
          <div
            className="circle5 white"
          />
          <div
            className="circle6 white"
          />
          <div
            className="circle7 white"
          />
          <div
            className="circle8 white"
          />
          <div
            className="circle9 white"
          />
          <div
            className="circle10 white"
          />
          <div
            className="circle11 white"
          />
          <div
            className="circle12 white"
          />
        </div>
      </span>
    </div>
  </div>
</div>
`;
